src/daft-local-execution/src/pipeline.rs

```diff
@@ -282,17 +283,63 @@ pub fn physical_plan_to_pipeline(
             group_by,
             pivot_column,
             value_column,
+            aggregation,
             names,
             ..
         }) => {
-            let pivot_op = PivotOperator::new(
+            let child_node = physical_plan_to_pipeline(input, psets, cfg)?;
+            // Pivot is a two stage operation. The first stage aggregates the values, grouped by both the group by columns and the pivot column.
+            // The second stage is a pivot operation where the values are pivoted into columns based on the pivot column.
+
+            let group_by_with_pivot = group_by
+                .iter()
+                .chain(std::iter::once(pivot_column))
+                .cloned()
+                .collect::<Vec<_>>();
+            let aggregate_schema = || -> DaftResult<_> {
+                let fields = group_by_with_pivot
+                    .iter()
+                    .map(|expr| expr.to_field(input.schema()))
+                    .chain(std::iter::once(aggregation.to_field(input.schema())))
+                    .collect::<DaftResult<Vec<_>>>()?;
+                Schema::new(fields)
+            }()
+            .with_context(|_| PipelineCreationSnafu {
+                plan_name: physical_plan.name(),
+            })?;
+            let (first_stage_aggs, second_stage_aggs, final_exprs) = populate_aggregation_stages(
+                &[aggregation.clone()],
+                &aggregate_schema.into(),
+                &group_by_with_pivot,
+            );
```

Comment:
Isn't the two-stage agg only necessary for partitioned distributed computations? To my knowledge, we do this 2-stage thing so that we can perform a local agg (as an optimization to reduce data cardinality) before the shuffle, and then perform the second local agg + final project to correctly execute the operation.

For local computations on swordfish, would it not be more performant to just perform a fully-materializing single stage aggregation before the pivot?

Reply:
It was actually faster to still use the two-stage agg pattern on swordfish, vs full-materialize then agg, so i kept it around. 

But, the simpler and way more effective way to do it local would be to do a `.fold` like pattern. This is something I plan on doing.

I probably shouldn't use the two stage for pivot then, will remove it and keep it simple.

Reply:
Modified it to be a fully-materializing single stage agg.